lib/tapioca/compilers/dsl/rails_generators.rb

```diff
@@ -0,0 +1,131 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "rails/generators"
+  require "rails/generators/app_base"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::RailsGenerators` generates RBI files for Rails generators
+      #
+      # For example, with the following generator:
+      #
+      # ~~~rb
+      # # lib/generators/sample_generator.rb
+      # class ServiceGenerator < Rails::Generators::NamedBase
+      #   argument :result_type, type: :string
+      #
+      #   class_option :skip_comments, type: :boolean, default: false
+      # end
+      # ~~~
+      #
+      # this compiler will produce the RBI file `service_generator.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # service_generator.rbi
+      # # typed: strong
+      #
+      # class ServiceGenerator
+      #   sig { returns(::String)}
```

Comment:
I'm not familiar with Rails generator but could this be nil? Or will the generator error before?

Reply:
Arguments are required, options are not. So this will always be a String, not `T.nilable(String)`. The only nilable DSL methods are `class_options` without a default.

To better understand the difference, arguments are akin to positional arguments while options are akin to keyword arguments. As an example:

    bin/rails g my_generator <argument> [--<class_option> [value]]

I could add an option without a default to demonstrate a nilable option as part of this if you'd like. It's covered by the test suite.